### PR TITLE
NodeJS v12 compatibility

### PR DESCRIPTION
**Pull request by goooseman**
`ref` package does not work with NodeJS v12 and will never be: [source](https://github.com/TooTallNate/ref/issues/125#issuecomment-595404687)

This PR updates `.nvmrc` to NodeJS 12 and uses `ref-napi` libs family instead of `ref` to resolve the issue.

Two type definitions are added to the project, which will be removed after [this](https://github.com/Janealter/ref-struct-napi/pull/1) and [this](https://github.com/Janealter/ref-array-napi/pull/1) PR's will be merged and released.

This change is tested with NodeJS 12.13.0, MacOS 10.13.6 and my Canon 80D. Everything works as before.

The example folder is a broken at the moment, because it attempts to install an old ts-gphoto2-driver version and it fails, because of `ref` module can not be compiled under NodeJS v12. I suggest to release a `2.4.0` version of this lib and bump the dependency version in `examples/package.json` after.

### CODE REVIEW

**Comment by goooseman** [comment]
The build is failing, because `examples/package.json` installs `"@typedproject/gphoto2-driver": "2.2.3"`, which uses `ref`, which fails to build under NodeJS v12.

Will be resolved after release, by bumping the dependency version in `examples/package.json`

File: @types/ref-array-napi/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,65 @@
+// Type definitions for ref-array-napi
```
Comment (Romakita) [inline @types/ref-array-napi/index.d.ts:1]: Good idea :) but, I have a doubt when the npm package will be released. Theses files won't be packaged. It means theses types won't be available when a developer install our package.

Comment (Romakita) [reply @types/ref-array-napi/index.d.ts:1]: I think to fix it, you have to move this files in src/@types and import it in index.ts? Why do you think ?

Comment (goooseman) [reply @types/ref-array-napi/index.d.ts:1]: I'll try it out right now.

**Comment by goooseman** [comment]
Ohhhh, I didn't see you forced push and I've pushed again.

Can you please make a force push again from your local branch without pulling the changes from remote? This will fix the issue.

**Comment by goooseman** [comment]
BTW, moving `@types` to `src/@types` and importing in `index.ts` works and sounds like a great idea

**Comment by goooseman** [comment]
Ok, I've resolved the git issue and moved `@types` to `src`

**Comment by Romakita** [comment]
Nice :). I'll merge your PR

**Comment by Romakita** [comment]
Release is on going :)

**Comment by Romakita** [comment]
Released :)

**Pull request merged**